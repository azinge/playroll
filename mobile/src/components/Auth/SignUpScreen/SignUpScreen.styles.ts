import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  mainContainer: {
    flex: 1,
    width: '100%',
    backgroundColor: 'white',
  },
  container: {
    flex: 1,
    width: '95%',
    paddingLeft: 20,
  },
  segueToSignInContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '22%',
    opacity: 0.5,
    paddingTop: 20,
  },
  signInTitle: {
    fontSize: 17,
    color: '#6A0070',
  },
  signupHeader: {
    flexDirection: 'row',
    width: '100%',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingTop: 30,
    paddingBottom: 20,
    paddingRight: 20,
  },
  signupTextContainer: {
    flex: 1,
    alignItems: 'center',
    right: 25,
  },
  signupText: {
    fontWeight: 'bold',
    fontSize: 30,
    color: '#6A0070',
  },
  imageSelectionContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    height: 75,
    width: 75,
    borderRadius: 75 / 2,
    backgroundColor: 'pink',
  },
  editPhotoText: {
    paddingTop: 10,
    fontSize: 15,
    color: '#6A0070',
  },
  formText: {
    fontSize: 15,
    color: '#6A0070',
  },
  inputContainer: {
    width: '100%',
    height: 45,
    marginBottom: 15,
    alignItems: 'center',
    justifyContent: 'center',
    borderBottomWidth: 1,
    borderColor: '#6A0070',
    color: '#6A0070',
  },
  showPasswordButton: {
    marginTop: 15,
  },
  submitButton: {
    flex: 1,
    position: 'absolute',
    left: 0,
    bottom: 0,
    width: '100%',
    backgroundColor: '#6A0070',
    padding: 25,
  },
  submitButtonText: {
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: 20,
  },
  errorMessage: {
    marginTop: 15,
    fontWeight: 'bold',
    fontSize: 18,
    color: '#FF0000',
  },
  tosLink: {
    fontWeight: 'bold',
    fontSize: 15,
    color: '#19B3E2',
  },
});
