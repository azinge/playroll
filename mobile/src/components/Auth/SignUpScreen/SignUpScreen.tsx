/**
 * SignUpScreen
 */

import * as React from 'react';
import {
  ActivityIndicator,
  Text,
  TextInput,
  TouchableOpacity,
  View,
  TouchableWithoutFeedback,
  Keyboard,
  Platform,
  Linking,
  SafeAreaView,
  Switch,
} from 'react-native';
import { WebBrowser } from 'expo';
import { NavigationScreenProp } from 'react-navigation';
import { SignUpMutation } from '../../../graphql/requests/Auth';
import styles from './SignUpScreen.styles';

export interface Props {
  toggleSignUp?: () => void;
  navigation?: NavigationScreenProp<{}>;
}

interface State {
  username: string;
  password: string;
  showPassword: boolean;
  email: string;
  avatar: string;
  error?: string;
}

export default class SignUpScreen extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      showPassword: true,
      avatar: '',
      error: undefined,
    };
    this.displayPassword = this.displayPassword.bind(this);
    this.renderError = this.renderError.bind(this);
  }

  displayPassword() {
    this.setState({ showPassword: !this.state.showPassword });
  }

  handleOpenTOSURL() {
    const url = 'http://www.playroll.io/tos';
    if (Platform.OS === 'ios') {
      WebBrowser.openBrowserAsync(url);
    }
    // Linking.openURL(url);
    WebBrowser.openBrowserAsync(url);
  }

  validateInput(signUp) {
    if (this.state.username === '' || this.state.email === '' ||
        this.state.password === '' || this.state.avatar === '') {
          return this.setState({
            error: 'All fields must have a value'
          }, () => {
            setTimeout(() => {this.setState({ error: null })}, 3000);
          });
    }
    signUp();
  }

  renderHeader() {
    return (
      <View style={styles.signupHeader}>
        <Text style={styles.signupText}>Sign Up</Text>
      </View>
    );
  }

  termsOfServiceLink() {
    return (
      <TouchableOpacity
        onPress={() => this.handleOpenTOSURL()}
        style={styles.tosContainer}
      >
        <Text style={styles.tosLink}>
          By signing up, you are agreeing to our Terms of Service.
        </Text>
      </TouchableOpacity>
    );
  }

  renderPasswordButton() {
    if (this.state.password.length > 1) {
      return (
        <TouchableOpacity
          onPress={() => this.displayPassword()}
          style={styles.showPasswordButton}
        >
          <Text style={styles.tosLink}>
            Show Password
          </Text>
        </TouchableOpacity>
      );
    }
  }

  renderSignupButton() {
    return (
      <SignUpMutation
        variables={{
          username: this.state.username,
          password: this.state.password,
          email: this.state.email,
          avatar: this.state.avatar,
        }}
      >
        {(signUp, { loading, error, data }) => {
          if (data) {
            this.props.navigation.navigate("Confirmation");
          }
          return (
            <TouchableOpacity
              onPress={() => this.validateInput(signUp)}
              style={styles.submitButton}
            >
              {loading ? (
                <ActivityIndicator color={'white'} />
              ) : (
                <Text style={styles.submitButtonText}>Sign Up</Text>
              )}
            </TouchableOpacity>
          );
        }}
      </SignUpMutation>
    );
  }

  renderError() {
    return <Text style={styles.errorMessage}>{this.state.error}</Text>;
  }

  render() {
    return (
      <TouchableWithoutFeedback
        onPress={() => {
          Keyboard.dismiss();
        }}
      >
        <SafeAreaView style={styles.mainContainer}>
          <View style={styles.container}>
            {this.renderHeader()}
            <TextInput
              placeholder='Username'
              autoCapitalize='none'
              style={styles.inputContainer}
              onChangeText={(username: string) => this.setState({ username })}
              autoCapitalize={'sentences'}
              value={this.state.username}
            />
            <TextInput
              placeholder='Email'
              style={styles.inputContainer}
              onChangeText={(email: string) => this.setState({ email })}
              autoCapitalize={'none'}
              value={this.state.email}
            />
            <TextInput
              placeholder='Password'
              style={styles.inputContainer}
              onChangeText={(password: string) => this.setState({ password })}
              secureTextEntry={this.state.showPassword}
              value={this.state.password}
            />
            {this.renderPasswordButton()}
            <TextInput // TODO: Remove Later
              style={styles.avatarContainer}
              autoCapitalize='none'
              placeholder='Avatar link'
              onChangeText={(avatar: string) => this.setState({ avatar })}
              value={this.state.avatar}
            />
            {this.termsOfServiceLink()}
            {this.renderError()}
          </View>
          {this.renderSignupButton()}
        </SafeAreaView>
      </TouchableWithoutFeedback>
    );
  }
}
