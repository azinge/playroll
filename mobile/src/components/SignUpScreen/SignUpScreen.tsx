/**
 * SignUpScreen
 */

import * as React from "react";
import {
  Text,
  SafeAreaView,
  View,
  TextInput,
  Button,
  Switch,
} from "react-native";

import { SIGN_UP_MUTATION, SignUpMutation } from "../../graphql/requests/Auth";
import {
  CONFIRM_SIGN_UP_MUTATION,
  ConfirmSignUpMutation,
} from "../../graphql/requests/Auth";
export interface Props {
  toggleSignUp: () => void;
}

interface State {
  username: string;
  password: string;
  email: string;
  confirmUser: string;
  authCode: string;
  showPassword: boolean;
}

export default class SignUpScreen extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      email: "",
      confirmUser: "",
      authCode: "",
      showPassword: true,
    };
    this.toggleSwitch = this.toggleSwitch.bind(this);
  }

  toggleSwitch() {
    this.setState({ showPassword: !this.state.showPassword });
  }
  render() {
    return (
      <View style={{ backgroundColor: "white", flex: 2 }}>
        <Text>SignUpScreen</Text>
        <TextInput
          style={{ height: 40, borderColor: "gray", borderWidth: 1 }}
          autoCapitalize="none"
          placeholder="Username"
          onChangeText={(username: string) => this.setState({ username })}
          value={this.state.username}
        />
        <View
          style={{
            flexDirection: "row",
            alignItems: "center",
            justifyContent: "center",
          }}
        >
          <TextInput
            secureTextEntry={this.state.showPassword}
            autoCapitalize="none"
            placeholder="Password"
            style={{ flex: 4, height: 40, borderColor: "gray", borderWidth: 1 }}
            onChangeText={(password: string) => this.setState({ password })}
            value={this.state.password}
          />
          <Switch
            onValueChange={this.toggleSwitch}
            value={!this.state.showPassword}
          />
        </View>
        <TextInput
          style={{ height: 40, borderColor: "gray", borderWidth: 1 }}
          autoCapitalize="none"
          placeholder="Email"
          onChangeText={(email: string) => this.setState({ email })}
          value={this.state.email}
        />
        <SignUpMutation
          mutation={SIGN_UP_MUTATION}
          variables={{
            username: this.state.username,
            password: this.state.password,
            email: this.state.email,
          }}
        >
          {(signUp, { data }) => {
            return (
              <Button
                title="Sign Up"
                onPress={() => {
                  signUp();
                }}
              />
            );
          }}
        </SignUpMutation>
        <TextInput
          style={{ height: 40, borderColor: "black", borderWidth: 1 }}
          autoCapitalize="none"
          placeholder="Username"
          onChangeText={(confirmUser: string) => this.setState({ confirmUser })}
          value={this.state.confirmUser}
        />
        <TextInput
          style={{ height: 40, borderColor: "black", borderWidth: 1 }}
          autoCapitalize="none"
          placeholder="Code"
          onChangeText={(authCode: string) => this.setState({ authCode })}
          value={this.state.authCode}
        />
        <ConfirmSignUpMutation
          mutation={CONFIRM_SIGN_UP_MUTATION}
          variables={{
            username: this.state.confirmUser,
            code: this.state.authCode,
          }}
        >
          {(confirmSignUp, { data }) => {
            return (
              <Button
                title="Confirm Sign Up"
                onPress={() => {
                  confirmSignUp();
                }}
              />
            );
          }}
        </ConfirmSignUpMutation>
      </View>
    );
  }
}
