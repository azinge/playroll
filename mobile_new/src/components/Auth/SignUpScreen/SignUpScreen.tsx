/**
 * SignUpScreen
 */

import * as React from 'react';
import {
  Text,
  SafeAreaView,
  View,
  TextInput,
  Button,
  Switch,
} from 'react-native';
import { NavigationScreenProp } from 'react-navigation';

import {
  SignUpMutation,
  ConfirmSignUpMutation,
} from '../../../graphql/requests/Auth';

import styles from './SignUpScreen.styles';
export interface Props {
  toggleSignUp?: () => void;
  navigation?: NavigationScreenProp<{}>;
}

interface State {
  username: string;
  password: string;
  email: string;
  avatar: string;
  confirmUser: string;
  authCode: string;
  showPassword: boolean;
}

export default class SignUpScreen extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      email: '',
      avatar:
        'https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/User_icon_2.svg/440px-User_icon_2.svg.png',
      confirmUser: '',
      authCode: '',
      showPassword: true,
    };
    this.toggleSwitch = this.toggleSwitch.bind(this);
  }

  toggleSwitch() {
    this.setState({ showPassword: !this.state.showPassword });
  }
  render() {
    return (
      <SafeAreaView style={styles.screenContainer}>
        <View style={styles.optionsContainer}>
          <Text>SignUpScreen</Text>
          <TextInput
            style={styles.usernameField}
            autoCapitalize='none'
            placeholder='Username'
            onChangeText={(username: string) => this.setState({ username })}
            value={this.state.username}
          />
          <View style={styles.passwordContainer}>
            <TextInput
              secureTextEntry={this.state.showPassword}
              autoCapitalize='none'
              placeholder='Password'
              style={styles.passwordField}
              onChangeText={(password: string) => this.setState({ password })}
              value={this.state.password}
            />
            <Switch
              onValueChange={this.toggleSwitch}
              value={!this.state.showPassword}
            />
          </View>
          <TextInput
            style={styles.emailField}
            autoCapitalize='none'
            placeholder='Email'
            onChangeText={(email: string) => this.setState({ email })}
            value={this.state.email}
          />
          <TextInput // TODO: Remove Later
            style={styles.emailField}
            autoCapitalize='none'
            placeholder='Avatar link'
            onChangeText={(avatar: string) => this.setState({ avatar })}
            value={this.state.avatar}
          />
          <SignUpMutation
            variables={{
              username: this.state.username,
              password: this.state.password,
              email: this.state.email,
              avatar: this.state.avatar,
            }}
          >
            {(signUp, { data }) => {
              return (
                <Button
                  title='Sign Up'
                  onPress={() => {
                    signUp();
                  }}
                />
              );
            }}
          </SignUpMutation>
          <TextInput
            style={styles.confirmUsernameField}
            autoCapitalize='none'
            placeholder='Username'
            onChangeText={(confirmUser: string) =>
              this.setState({ confirmUser })
            }
            value={this.state.confirmUser}
          />
          <TextInput
            style={styles.confirmCodeField}
            autoCapitalize='none'
            placeholder='Code'
            onChangeText={(authCode: string) => this.setState({ authCode })}
            value={this.state.authCode}
          />
          <ConfirmSignUpMutation
            variables={{
              username: this.state.confirmUser,
              code: this.state.authCode,
            }}
          >
            {(confirmSignUp, { data }) => {
              return (
                <Button
                  title='Confirm Sign Up'
                  onPress={() => {
                    confirmSignUp();
                  }}
                />
              );
            }}
          </ConfirmSignUpMutation>
        </View>
      </SafeAreaView>
    );
  }
}
