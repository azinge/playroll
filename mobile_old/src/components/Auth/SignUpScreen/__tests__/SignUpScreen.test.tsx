import * as React from "react";
import renderer from "react-test-renderer";
import { MockedProvider } from "react-apollo/test-utils";

import SignUpScreen from "../SignUpScreen";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MockedProvider>
        <SignUpScreen />
      </MockedProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
