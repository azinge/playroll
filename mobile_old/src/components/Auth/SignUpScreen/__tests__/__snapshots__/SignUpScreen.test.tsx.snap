// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 2,
      }
    }
  >
    <Text>
      SignUpScreen
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Username"
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "flex": 4,
            "height": 40,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <RCTSwitch
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        value={false}
      />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Avatar link"
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/User_icon_2.svg/440px-User_icon_2.svg.png"
    />
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Sign Up
        </Text>
      </View>
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Username"
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Code"
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Confirm Sign Up
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
